src/Mod/Part/App/TopoShapeExpansion.cpp

```diff
@@ -120,6 +120,48 @@ void TopoShape::initCache(int reset) const
     }
 }
 
+Data::ElementMapPtr TopoShape::resetElementMap(Data::ElementMapPtr elementMap)
+{
+    if (_cache && elementMap != this->elementMap(false)) {
+        for (auto& info : _cache->shapeAncestryCache) {
+            info.clear();
+        }
+    }
+    else {
+        initCache();
+    }
+    if (elementMap) {
+        _cache->cachedElementMap = elementMap;
+        _cache->subLocation.Identity();
+        _subLocation.Identity();
+        _parentCache.reset();
+    }
+    return Data::ComplexGeoData::resetElementMap(elementMap);
+}
+
+void TopoShape::flushElementMap() const
+{
+    initCache();
+    if (!elementMap(false) && this->_cache) {
+        if (this->_cache->cachedElementMap) {
+            const_cast<TopoShape*>(this)->resetElementMap(this->_cache->cachedElementMap);
+        }
+        else if (this->_parentCache) {
+            TopoShape parent(this->Tag, this->Hasher, this->_parentCache->shape);
+            parent._cache = _parentCache;
+            parent.flushElementMap();
+            TopoShape self(this->Tag,
+                           this->Hasher,
+                           this->_Shape.Located(this->_subLocation * this->_cache->subLocation));
+            self._cache = _cache;
+            self.mapSubElement(parent);
+            this->_parentCache.reset();
+            this->_subLocation.Identity();
+            const_cast<TopoShape*>(this)->resetElementMap(self.elementMap());
+        }
+    }
+}
+
```

Comment:
Why duplicating what's already in https://github.com/FreeCAD/FreeCAD/pull/12471?

Reply:
Because these have to exist to proceed with anything here, and #12471 has never come out of draft.  Doesn't matter to me how they make it in as long as they make it in very soon.

Reply:
Not the correctness of the = operator.   ALL tests are suspect unless these are in place, which means it is a blocker to any progress on anything with a test - I.E. any work at all on Toponaming.